web/gui/dashboard/src/domains/chart/components/chart-container.tsx

```diff
@@ -0,0 +1,202 @@
+import { forEachObjIndexed } from "ramda"
+import React, { useEffect, useState, useLayoutEffect } from "react"
+import { useSelector, useDispatch } from "react-redux"
+import { useInterval, useThrottle } from "react-use"
+
+import { AppStateT } from "store/app-state"
+
+import { selectGlobalPanAndZoom } from "domains/global/selectors"
+
+import { chartLibrariesSettings } from "../utils/chartLibrariesSettings"
+import { Attributes } from "../utils/transformDataAttributes"
+import { getChartPixelsPerPoint } from "../utils/get-chart-pixels-per-point"
+import { getPortalNodeStyles } from "../utils/get-portal-node-styles"
+
+import { fetchDataAction } from "../actions"
+import { selectChartData, selectChartDetails, selectChartFetchDataParams } from "../selectors"
+
+import { Chart } from "./chart"
+
+const getChartURLOptions = (attributes: Attributes) => {
+  const {
+    appendOptions,
+    overrideOptions,
+  } = attributes
+  let ret = ""
+
+  ret += overrideOptions
+    ? overrideOptions.toString()
+    : chartLibrariesSettings.dygraph.options(attributes)
+
+  if (typeof appendOptions === "string") {
+    ret += `|${encodeURIComponent(appendOptions)}`
+  }
+
+  ret += "|jsonwrap"
+
+  // todo
+  const isForUniqueId = false
+  // always add `nonzero` when it's used to create a chartDataUniqueID
+  // we cannot just remove `nonzero` because of backwards compatibility with old snapshots
+  if (isForUniqueId || window.NETDATA.options.current.eliminate_zero_dimensions) {
+    ret += "|nonzero"
+  }
+
+  return ret
+}
+
+
+export type Props = {
+  attributes: Attributes
+  chartUuid: string
+  portalNode: HTMLElement
+}
+
+export const ChartContainer = ({
+  attributes,
+  chartUuid,
+  portalNode,
+}: Props) => {
+  const chartDetails = useSelector((state: AppStateT) => selectChartDetails(
+    state, { id: chartUuid },
+  ))
+
+  const [shouldFetch, setShouldFetch] = useState<boolean>(true)
+  useInterval(() => {
+    setShouldFetch(true)
+  }, 2000)
```

Comment:
2000 could be a named constant / config member somewhere for easier control

Reply:
thx for reminder, forgot to place `todo` here, the interval will be set based on settings from BE